src/components/visualizations/chart/chartOptions/extendedStackingChartOptions.ts

```diff
@@ -36,21 +37,33 @@ export function getCategoriesForTwoAttributes(
             parentAttr: Execution.IResultAttributeHeaderItem,
             index: number,
         ) => {
-            const key: string = get(parentAttr, "attributeHeaderItem.name", "");
+            const uri: string = get(parentAttr, "attributeHeaderItem.uri", "");
+            const name: string = get(parentAttr, "attributeHeaderItem.name", "");
             const value: string = get(children[index], "attributeHeaderItem.name", "");
 
-            const childCategories: string[] = result[key] || [];
+            const childCategories: string[] = get(result, `${uri}.categories`, []);
+
+            if (!childCategories.length) {
+                keys.push(uri);
+            }
 
             return {
                 ...result,
-                [key]: [...childCategories, value], // append value
+                [uri]: {
+                    name,
+                    categories: [...childCategories, value], // append value
+                },
             };
         },
         {},
     );
 
-    return Object.keys(combinedResult).map((name: string) => ({
-        name,
-        categories: combinedResult[name],
-    }));
+    return keys.map((key: string) => {
```

Comment:
I'm just wondering if we need this `keys` array. I don't have a strong argument, but I feel that having .reduce() with side effects (pushing to keys array) is something we should not do.

What about just use `values` from lodash (https://lodash.com/docs/#values):
```
return values(combinedResult);
```
Then, the reduce would be without side effects and the keys array would not be needed.

Reply:
The root cause of this bug comes from `Object.keys` issue, it returns an unexpected order. Assume we have an object like following:
```
const combinedResult = {
   "3": {},
   "2": {},
  "1": {}
};

Object.keys(combinedResult); // it returns ["1", "2", "3"] but the expected order is ["3", "2", "1"]
```
And unfortunately, `Object.values` or `values` from lodash that will return wrong order too.
That why I need `keys` array

Reply:
Thanks for explanation